pkg/util/util.go

```diff
@@ -129,3 +132,33 @@ func UnArchiveLocalTar(filePath, destDir string, arg ...string) error {
 	fileReader := bufio.NewReader(file)
 	return UnArchiveTar(fileReader, destDir, arg...)
 }
+
+// CopyFile copies a file from one location to another.
+func CopyFile(src, dst string) error {
+	in, err := os.Open(src)
+	if err != nil {
+		return err
+	}
+	defer in.Close()
+
+	out, err := os.Create(dst)
+	if err != nil {
+		return err
+	}
+	defer out.Close()
+
+	_, err = io.Copy(out, in)
+	if err != nil {
+		return err
+	}
+	return out.Close()
+}
+
+// MoveFile moves a file by doing a copy and then removing the source file. This works across file system which os.Rename won't
+func MoveFile(src, dst string) error {
+	err := CopyFile(src, dst)
```

Comment:
For files that are located in the same file system we would like to have actual Rename and not copy - much faster.
So, it is worth executing rename and fallback to copy if fails

Reply:
This is a helper function explicitly for when the files are not on the same file system, I guess I should name the function more explicit that this is the case. We don't really want to encourage people calling this unless they know for a fact that os.Rename doesn't work.